lucene/core/src/java/org/apache/lucene/index/CheckIndex.java

```
@@ -2585,10 +2585,7 @@ public static Status.VectorValuesStatus testVectors(
                       + "\" has vector values but dimension is "
                       + dimension);
             }
-            VectorValues values = reader.getVectorValues(fieldInfo.name);
-            if (values == null) {
-              continue;
-            }
+            VectorValues values = VectorValues.getVectorValues(reader, fieldInfo.name);
```

`values == null` should always return false since we checked the field infos at that point already, so maybe we should throw an exception instead of doing a "continue"? (or do vectors have ghost fields? I thought they would guarantee that seeing vectors on the field infos would guarantee a non-null vector values)

---

Umm a bit confused here as we have removed the null check now since we don't return null in `#VectorValues.getVectorValues` so where should we throw an exception. Do you mean to not change this and update the null check to throw an IAE exception?

---

> Do you mean to not change this and update the null check to throw an IAE exception?

Yes, though maybe a `CheckIndexException` rather than an IAE.

---

Got it thanks💯.